js/index.js

```diff
@@ -0,0 +1,296 @@
+        }
+    });
+});
+
+// Recipe card creation
+function createRecipeCard(recipe) {
+    return `
+        <div class="recipe-card" data-recipe-id="${recipe.id}">
+            <div class="recipe-image">${recipe.icon}</div>
+            <div class="recipe-info">
+                <div class="recipe-category">${recipe.category}</div>
+                <h3 class="recipe-title">${recipe.title}</h3>
+                <p class="recipe-description">${recipe.description}</p>
+                <div class="recipe-meta">
+                    <span class="recipe-time">⏱️ ${recipe.time}</span>
+                    <span class="recipe-rating">★ ${recipe.rating}</span>
+                </div>
+            </div>
+        </div>
+    `;
+}
+
+// Display recipes in grid
+function displayRecipes(recipes) {
+    recipesGrid.innerHTML = recipes.map(recipe => createRecipeCard(recipe)).join('');
+    
+    // Add click handlers to recipe cards
+    document.querySelectorAll('.recipe-card').forEach(card => {
+        card.addEventListener('click', () => {
+            const recipeId = card.getAttribute('data-recipe-id');
+            showRecipeModal(recipeId);
+        });
+    });
+}
+
+// Recipe modal functionality (placeholder for future development)
+function showRecipeModal(recipeId) {
+    const recipe = sampleRecipes.find(r => r.id == recipeId);
+    if (recipe) {
+        // For now, show an alert. In the full application, this would open a detailed recipe modal
+        alert(`Recipe Details:\n\nTitle: ${recipe.title}\nCategory: ${recipe.category}\nTime: ${recipe.time}\nRating: ${recipe.rating}\n\nDescription: ${recipe.description}\n\nNote: Full recipe details will be available after signing in!`);
+    }
+}
+
+// Load more button functionality
+loadMoreBtn.addEventListener('click', () => {
+    // In a real application, this would load more recipes from the server
+    alert('More recipes would be loaded here! Please sign in to access the full recipe collection.');
+});
+
+// Animation on scroll for recipe cards
+function animateOnScroll() {
+    const cards = document.querySelectorAll('.recipe-card, .feature-card');
+    const observer = new IntersectionObserver((entries) => {
+        entries.forEach(entry => {
+            if (entry.isIntersecting) {
+                entry.target.style.opacity = '1';
+                entry.target.style.transform = 'translateY(0)';
+            }
+        });
+    }, { threshold: 0.1 });
+
+    cards.forEach(card => {
+        card.style.opacity = '0';
+        card.style.transform = 'translateY(30px)';
+        card.style.transition = 'opacity 0.6s ease, transform 0.6s ease';
+        observer.observe(card);
+    });
+}
+
+// Initialize the page
+document.addEventListener('DOMContentLoaded', () => {
+    // Display initial recipes
+    displayRecipes(sampleRecipes);
+    
+    // Set up scroll animations
+    animateOnScroll();
+    
+    // Add entrance animation to hero elements
+    setTimeout(() => {
```

Comment:
would be good to add a comment on how setTimeout works:
`// using setTime out to delay content delivery for UX`